src/app/customer-search/customer-search.component.ts

```diff
@@ -12,62 +12,71 @@ import { TfiDataService } from '../services/tfi-data.service';
 export class CustomerSearchComponent implements OnInit {
   public idSearch: string;
   public searchBtn: boolean = true; // Search button is disabled while querying DB
+  public invalidID: boolean = false;
   public resultsTable = [];
   public currentPage: number = 0;
 
   constructor(
-    private testDataService: TestsDataService, private tsDataService: TsScreenerDataService, private thsDataService: ThsDataService, private tfiDataService: TfiDataService)
-    { }
+    private testDataService: TestsDataService, private tsDataService: TsScreenerDataService, private thsDataService: ThsDataService, private tfiDataService: TfiDataService) { }
 
-  ngOnInit() {
+  public ngOnInit() {
   }
 
   // Gets JSON from queryDB() puts it into an array to create the table of appointments
   // sorts by date, most recent first.
   public patientSearch() {
+    if (!this.validID()) { this.invalidID = true; return; };
+    this.invalidID = false;
     this.searchBtn = false;
     let appts: Object[] = this.queryDB();
     this.currentPage = 0;
-    this.resultsTable = appts.map(x => Object.assign({}, x));
-    this.resultsTable.sort((a,b) => {
+    this.resultsTable = appts.map((x) => Object.assign({}, x));
+    this.resultsTable.sort((a, b) => {
       let date1 = new Date(a.date);
       let date2 = new Date(b.date);
-      if(date1.valueOf() < date2.valueOf()) {
+      let diff = date2.valueOf() - date1.valueOf();
+      if (diff > 0) {
```

Comment:
You can just `return diff`; the value doesn't need to be 1/0/-1, just positive/0/negative.